bin/c8.js

```diff
@@ -21,12 +21,17 @@ async function run () {
   ].indexOf(argv._[0]) !== -1) {
     argv = buildYargs(true).parse(process.argv.slice(2))
   } else {
+    // allow c8 to run on Node 8 (coverage just won't work).
+    if (!promises) {
```

Comment:
maybe just pull in `semver` a dependency and:

```
if (semver.lt('10.12.0', process.version)) {
  console.warn('native test coverage was not added to V8 until 10.12.0')
  foreground(hideInstrumenterArgs(argv))
  return
}
```

that way we're not relying on `util.promisify` not being back-ported.

👍 thanks for your work on this; I'll be back a bit later today.

Reply:
> that way we're not relying on `util.promisify` not being back-ported.

To be accurate, the API used for feature detection here is not `util.promisify` but [`fs.promises`](https://nodejs.org/api/fs.html#fs_fs_promises_api).

I can hardly imagine the Node.js team decides to back-port `fs.promises` to [the maintenance LTS version](https://github.com/nodejs/Release/tree/87c98116e471e86a66dc39ebe703d3394323f8da#release-schedule).

Reply:
perhaps just add a comment then:

```
// fs.promises was not added until Node v10.0.0, if it doesn't
// exist, assume we're Node v8.x and skip coverage.
```

Reply:
938a46391105b4f56bc1c07144dba362da74b4b1